usecases/cluster/conflict_delegate.go

```diff
@@ -0,0 +1,83 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2024 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package cluster
+
+import (
+	"os"
+
+	"github.com/hashicorp/memberlist"
+	"github.com/hashicorp/raft"
+	"github.com/sirupsen/logrus"
+)
+
+type conflictDelegate struct {
+	logger  logrus.FieldLogger
+	localID string
+	raft    *raft.Raft
+	voter   bool
+}
+
+func (d *conflictDelegate) SetVoter(voter bool) {
+	d.voter = voter
+}
+
+func (d *conflictDelegate) SetRaft(raft *raft.Raft) {
+	d.raft = raft
+}
+
+// NotifyConflict is invoked when a name conflict is detected
+func (d *conflictDelegate) NotifyConflict(existing, other *memberlist.Node) {
+	if d.raft == nil {
+		d.logger.WithFields(logrus.Fields{
+			"name":        existing.Name,
+			"existing_ip": existing.Address(),
+			"new_ip":      other.Address(),
+		}).Warn("raft is not up yet")
+		return
+	}
+
+	if existing.Name == d.localID {
+		d.logger.WithFields(logrus.Fields{
+			"name":        existing.Name,
+			"existing_ip": existing.Address(),
+			"new_ip":      other.Address(),
+		}).Warn("node conflicting IPs, i will shutdown ...")
+
+		// we force exit here for immediate stop of the node to avoid any raft replication.
+		os.Exit(1)
+	}
+
+	_, leaderID := d.raft.LeaderWithID()
```

Comment:
What if an election is ongoing ? We still want the change to apply no matter what. Maybe we could change it (in raft) so that on becoming the leader we do a pass to check that the IP <-> node-id between memberlist and raft are aligned ? Wdyt ?

Reply:
if the election is going `leaderID` will be empty , it's no op because the following ops are leader only